src/appleseed/foundation/utility/unzipper.cpp

```diff
@@ -163,7 +162,8 @@ void unzip(const string& zip_filename, const string& unzipped_dir)
 
 bool has_extension(const string& filename, const string& extension) 
 {
-    return filename.rfind(extension) == filename.size() - extension.size();
+    size_t extension_start = filename.rfind(extension);
+    return  extension_start != string::npos && extension_start == filename.size() - extension.size();
```

Comment:
Actually, now that I think of it, I suggest getting rid of the whole function and using `foundation::ends_with()` instead. Sorry, I didn't think of that before :)